### ISSUE TITLE
CA2000 with SafeHandle

### ISSUE THREAD

**Issue by NN---** [dotnet/roslyn-analyzers#4457]
### Analyzer

**Diagnostic ID**: [CA2000](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000)

### Describe the improvement

Validating SafeHandle before `using` should not issue CA2000.

```cs
    class MySafeHandle : SafeHandleZeroOrMinusOneIsInvalid
    {
        public MySafeHandle(IntPtr s) : base(true) { }
        protected override bool ReleaseHandle() => true;
    }

    static class MySafeHandleGetter
    {
        public static bool TryGetMySafeHandle(out MySafeHandle? value)
        {
            value = new MySafeHandle(IntPtr.Zero);
            return true;
        }
    }

    class T
    {
        public void F()
        {
            if (!MySafeHandleGetter.TryGetMySafeHandle(out MySafeHandle? handle))
            {
                return;
            }

            // Effectively Dispose performs these checks inside but I do want to do them manually.
            // using (handle) { ... }

            // Validate handle
            if (handle == null || handle.IsInvalid)
            {
                return;
            }

            using (handle)
            {
                // Use handle and Dispose
                Console.WriteLine(handle.DangerousGetHandle());
            }
        }
    }
```
